s29986-PrzemyslawKosiedowski/zajecia2/zadanie2.txt

```diff
@@ -0,0 +1,81 @@
+Ćwiczenie 2
+===============================================================================
+
+To zadanie należy wykonać wpisując polecenia w terminalu normalnie. Nie należy
+tworzyć skryptu z tymi poleceniami.
+
+Zobacz taką serię poleceń:
+
+X=tekst
+echo $X
+bash
+echo $X
+
+Co się stało? Czy potrafisz wyjaśnić?
+-------------------------------------------------------------------------------
+
+Odpowiedź:
+X=tekst - do zmiennej X przypisuję słowo tekst
+echo $X - wyświetlam  zawartość zmiennej X - wypisuję "tekst"
+bash - uruchamiam nową instancję powłoki bash w bieżącym terminalu
+echo $X - ponownie wyświetlam zawartość zmiennej X - tym razem nic nie
+wypisuję. Dzieje się tak, ponieważ nowa instancja basha nie ma dostępu do
+zmiennych w starej powłoce.
+
+Wniosek:
+To zadanie nauczyło mnie, że po utworzeniu nowej powłoki nie mamy dostępu do
+zmiennych utworzonych na wcześniejszej powłoce. Można to wykorzystać np. do
+testowania skryptów.
+
+
+===============================================================================
+Zobacz:
+
+X=Tekst dłuższy
+echo $X
+
+Co jest nie tak? jak to naprawić? Zobacz czy pomoże zastosowanie cudzysłowów i
+apostrofów.
+-------------------------------------------------------------------------------
+
+Wynik:
+Po wpisaniu "X=Tekst dłuższy" bash wyrzucił komunikat: 
+bash: dłuższy: nie znaleziono polecenia
+Po wpisaniu "echo $X" znowu nic się nie pokazało.
```

Comment:
To nie echo miało być w cudzysłowiu tylko wartość przekazana do zmiennej do echo czyli
X="Tekst dłuższy"